joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -934,6 +957,13 @@ controller_interface::CallbackReturn JointTrajectoryController::on_configure(
 controller_interface::CallbackReturn JointTrajectoryController::on_activate(
   const rclcpp_lifecycle::State &)
 {
+  // Setup time_data buffer used for scaling
+  TimeData time_data;
+  time_data.time = get_node()->now();
+  time_data.period = rclcpp::Duration::from_nanoseconds(0);
+  time_data.uptime = get_node()->now();
+  time_data_.initRT(time_data);
```

Comment:
or
```suggestion
  time_data_.writeFromNonRT(time_data);
```
I guess? If this is wrong, I have to change the other buffers in on_activate

Reply:
The [docs](https://control.ros.org/master/doc/ros2_controllers/doc/writing_new_controller.html) state

>  This method is part of the real-time loop, therefore avoid any reservation of memory and, in general, keep it as short as possible.

so, probably  `initRT` is the correct approach?

Reply:
possibly? but is a `reset()` before necessary then?